framework/include/outputs/Checkpoint.h

```diff
@@ -54,6 +54,14 @@ class Checkpoint : public FileOutput
    */
   Checkpoint(const InputParameters & parameters);
 
+  /// Method writing MeshMetaData to file. This method is static public so it
+  /// can be called from other classes, thus avoding duplicated code.
+  static void writeMeshMetaData(const processor_id_type pid,
```

Comment:
I feel this function should be a normal function in `MooseApp`. `MooseApp` owns the data. All places calling this function can see `MooseApp`.

Reply:
for me it's the same, @permcody suggested to add it in `Checkpoint` as it is part of writing a checkpoint. But you might be right... I'll let you too decide

Reply:
That is also my impression first so pretty sure Cody said that before checking the details of the codes.

Reply:
@YaqiWang Is correct registering Mesh meta data is done in the MooseApp. You just need to call MooseApp::registerRestartableData and supply MooseApp::MESH_META_DATA for the last item in the function.

https://github.com/idaholab/moose/blob/8bd5ca2c4d878f8ba48b6ac38a53de0ff348e0df/framework/include/base/MooseApp.h#L521

Reply:
SplitMeshAction is already writing to the data: https://github.com/idaholab/moose/blob/8bd5ca2c4d878f8ba48b6ac38a53de0ff348e0df/framework/src/actions/SplitMeshAction.C#L103

Reply:
> SplitMeshAction is already writing to the data:
> 
> https://github.com/idaholab/moose/blob/8bd5ca2c4d878f8ba48b6ac38a53de0ff348e0df/framework/src/actions/SplitMeshAction.C#L103

Yes, this was added by @YaqiWang a few weeks ago via #16218 so we can close this

Reply:
Just noticed how old all this was, the stale bot moved it my notification so I was thinking this was all new. Need to look at dates a bit better. 😆

Reply:
@aeslaughter yes this was old, however, I still need your help with #15972.which is the step after this. I don't really get what I'm doing wrong and why I'm not able to relaod data.